content/ADR-183-favorite-places.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: adr
+adr: 183
+title: Favorite Places
+date: 2023-02-07
+status: Review
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - davidejensen
+---
+
+## Abstract
+
+This document describes an approach on how to integrate the new Favorite Places functionality in Decentraland.
+
+## Glossary
+
+- **Place** A specific parcel or set of parcels identified by a [place UUID](/adr/ADR-186)
+- **Places** The structure of database and endpoints used in the places website
+
+## Problem
+
+To enhance the discoverability of the different Decentraland places, there is the need of adding a favorite mechanism. This will allow user to store and retrieve easily a set of their best loved places.
+To integrate this changes at multiple levels are required, from the visual representation in the explorer, to the backend structure that will allow the saving and retrieving of a user favorites.
+
+Following a list of the expected functionalities:
+
+- Users can add a place to their favorite from the explore menu
+- Users can navigate to a new tab listing all the places they added to favorites
+In a future iteration it will be possible to favorite places from the minimap HUD and also from the map parcel info pop-up, currently not possible due to technical limitations of the renderer map handling and Places API restriction to curated places.
+More in the limitations section.
+
+## Solution
+
+### Data storing
+
+After an initial discussion about whether to store favorites in the user profile or in Places, we are more incline to store everything in Places, more informations about the decision journey at the end.
+
+### Favorite retrieval
+
+The Places API will provide the required informations through:
+
+- An endpoint to obtain the (paginated if we decide for a big maximum amount) list of favorites.
+- An endpoint to add a new favorite for the current user (already available)
+- An endpoint to remove a favorite for the current user (already available)
+In additon a change will be required in the current places data structure in order to show the favorite information in the explore places list with a boolean. (already provided by the places API, but requires changes in browser-interface and unity-renderer)
+
+### Data structure
+
+To facilitate the addition of the feature when handling favorites we should use the same data structure currently used for Places, with the boolean addition previously mentioned:
```

Comment:
Is it really required? Maybe it can be handled by the backend depending on the endpoint call

Reply:
Currently places displayed in the explore menu are obtained from the lambda for hot scenes `/lambdas/explore/hot-scenes`, this returns a specific data structure with some info about realms and friends connected to that place. 

Because this info is retrieved from the browser-interface (kernel), we could add just the new boolean field `isFavorite` and obtain that from the places API.

Reply:
I meant just the `isFavorite` field, if places API has specific endpoints for that maybe we can avoid adding it